### ISSUE TITLE
Add TypeScript types for `@carbon/icons`

### ISSUE THREAD

**Issue by artemkolotilkin** [carbon-design-system/carbon#13143]
Since `@carbon/icons-angular` package is outdated and no longer maintained, I'm trying to use `@carbon/icons` package in my Angular project. Unfortunately, Angular complains that types are missing. It would be great to have them added.

**Comment by tay1orjones** [comment]
Thanks for reporting. Types are slowly being added by the community, focusing on the react components first #12513. 

Icons haven't really been looked at or discussed yet, but an idea for `@carbon/icons-react` is up at https://github.com/carbon-design-system/carbon/pull/12034

**Comment by fronbasal** [comment]
The [`carbon-icons-angular`](https://github.com/carbon-design-system/carbon-icons-angular#stop_sign-carbonicons-angular---deprecated-stop_sign) docs show how to implement `@carbon/icons` package in Angular ([codesandbox link](https://codesandbox.io/s/gracious-diffie-7khcp3?file=/src/app/app.module.ts)).

**Comment by adamalston** [comment]
Was this issue addressed by https://github.com/carbon-design-system/carbon/pull/14714?

**Comment by blikblum** [comment]
> Was this issue addressed by [#14714](https://github.com/carbon-design-system/carbon/pull/14714)?

No. It add types to @carbon/react/icons but not for @carbon/icons. Using latest version 11.83.0

I'm using carbon web components
